### DISCUSSION TITLE
API To change  transaction fails to find destination_id

### DISCUSSION THREAD

**Discussion by soloam** [firefly-iii/firefly-iii#8750] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

When I do a PUT to update a trsnasaction I inform 

`
{
  "apply_rules": false,
  "fire_webhooks": false,
  "transactions": [
    {
      "type": "transfer",
      "source_id": 7,
      "destination_id": 13
    }
  ]
}
`

This fails with error:

`
{
  "message": "Não foi possível encontrar uma conta de destino válida ao pesquisar pelo ID \"13\" ou nome \"\". (and 1 more error)",
  "errors": {
    "transactions.0.destination_id": [
      "Não foi possível encontrar uma conta de destino válida ao pesquisar pelo ID \"13\" ou nome \"\"."
    ],
    "transactions.0.destination_name": [
      "Não foi possível encontrar uma conta de destino válida ao pesquisar pelo ID \"13\" ou nome \"\"."
    ]
  }
}
`

Stating that it cant find a destination account with the number 13.

But If I supply the account name destination_name, it works and the transaction is done to the account with the id 13

Thank You

### Debug information

`
{
  "message": "Não foi possível encontrar uma conta de destino válida ao pesquisar pelo ID \"13\" ou nome \"\". (and 1 more error)",
  "errors": {
    "transactions.0.destination_id": [
      "Não foi possível encontrar uma conta de destino válida ao pesquisar pelo ID \"13\" ou nome \"\"."
    ],
    "transactions.0.destination_name": [
      "Não foi possível encontrar uma conta de destino válida ao pesquisar pelo ID \"13\" ou nome \"\"."
    ]
  }
}
`

### Expected behaviour

It sould assume the destination_id 13 and do the update

### Steps to reproduce

1. Make a withdrawal transfer
2. In the API PUT /api/v1/transactions/<<id>> call with:
`
{
  "apply_rules": false,
  "fire_webhooks": false,
  "transactions": [
    {
      "type": "transfer",
      "source_id": <<source id>>,
      "destination_id": <<destination id>>
    }
  ]
}
`

**Comment by JC5** [comment]
You can't convert a transaction into a transfer like that. So this is not a bug.

**Reply by soloam** [reply]
No? But if I do with destination_name instead destination_id it works and changes the type to transfer!

**Reply by JC5** [reply]
It doesn't when I do it, it remains a withdrawal. Even though it does, mistakenly, make a transaction between two asset accounts. It keeps calling it a withdrawal. I'll pick it up!

Edit: for another transaction, it does. Very strange!

**Comment by JC5** [comment]
https://github.com/firefly-iii/firefly-iii/issues/6868

**Comment by soloam** [comment]
I'm using this a lot, I detect with a script if a withdrawal is a transfer, and I change it with the api! It works ok, but I have to inform the destination_name, because if I do it with destination_id it does not work

**Reply by JC5** [reply]
And it's not supposed to, lol. It's buggy and I'm working on another solution. This "feature" will stop working in the next release.

**Reply by soloam** [reply]
Will it stop working but will it have a alternative way to do it? Because I depend a lot on this to manage my transactions!

**Reply by JC5** [reply]
Absolutely, but it will be a new endpoint that doesn't exist yet. Sorry about that.

**Reply by soloam** [reply]
No problem, I'll adjust my script to call the new endpoint! Tks
